### ISSUE TITLE
Fix-Timeout.patch broke multi line output

### ISSUE THREAD

**Issue by kokarez** [facebook/ds2#316]
Fix-Timeout use the table output as 1 entry per 1 stderr line read. When the process end, we call update_status.

```python
    while True:
        line = process.stderr.readline()
        output.append(line)  # <-- line by line
        if line == '' and process.poll() is not None:
            break
        if 'PASS:' in line or 'FAIL:' in line:
            print '\r' + line,
            update_print(name)
        elif 'RESULT:' in line:
           output = [line]
     exit_status = process.returncode
     passes, failures = parse_test_results(output)
     update_status(name, command, output if exit_status != 0 else None)
```

The problem here is that update_status use the output table as the return of process.communicate, which mean the entry[0] is stdout and the entry[1] is stderr.

```python
def update_status(name = None, command = None, output = None):
    global output_lock, test_counter, total_tests
    with output_lock:
        if output is not None:
            print >> sys.stderr
            print >> sys.stderr, "Failed test suite: %s" % name
            print >> sys.stderr, "Command invoked: %s" % ' '.join(command)
            print >> sys.stderr, "stdout:\n%s" % output[0]#  <-- stdout
            print >> sys.stderr, "stderr:\n%s" % output[1]#  <-- stderr
        sys.stderr.write("\r%*d out of %d test suites processed" %
            (len(str(total_tests)), test_counter.value, total_tests))
        test_counter.value += 1
```

**Comment by fjricci** [comment]
Is this breaking multi-line debug logging? Or causing test failures?

**Comment by kokarez** [comment]
It diesn't cause any failure. But in case of failure due to backtrace, only
the two first line will be print. The first line as stdout and the second
as stderr

**Comment by kokarez** [comment]
I found other problem with this patch. I don't know why (most likely due to my impl on macosx) but a lot of tests hang. But when they do, the tester hangs too.

I have done an other impl of this patch, which is still horrible, but detect timeout.

before: https://travis-ci.org/facebook/ds2/jobs/128268942
after: https://travis-ci.org/facebook/ds2/jobs/128468486

```python
    while True:
         line = process.stderr.readline()
         rfds, wfds, efds = select.select( [process.stderr], [], [], 20 )
         if not rfds:
             output = ['\nFAIL: timeout -> %s' % name, '']
             break
         line = rfds[0].readline()
         output.append(line)
```

**Comment by fjricci** [comment]
Can you upload a PR with your fix? Thanks!

**Comment by fjricci** [comment]
Oh, I see it inside of your OSX PR

**Comment by kokarez** [comment]
Yea but that an ugly version :) Let me some time to work on a clean one. Thanks!

**Comment by fjricci** [comment]
Fixed by @kokarez in 6e6591d
